### ISSUE TITLE
Braced block in node-name position?

### ISSUE THREAD

**Issue by gbj** [stoically/syn-rsx#10]
I'm working on use `syn-rsx` in a framework where it would be really useful to be able to write code like

`<{ ... some expression ... }/>`

At present this gives the error "[invalid tag name](https://github.com/stoically/syn-rsx/blob/59cdd248141472a2f0f65348a40753ef971999e6/src/parser.rs#L333)."

Let me know if you can think of any reason it wouldn't be possible or would require major changes, and if not I'd be happy to work on a PR for it.

**Comment by stoically** [comment]
Given that `NodeName` is an Enum, I think it should be fairly straight-forward to implement something like `NodeName::Block`, and I'd be happy to review a PR for that.

I'm curious, what's the use case?

**Comment by gbj** [comment]
So this turned out to be really easy to do and works in the library I was going to use it for as intended. Kudos on a well-written and easy-to-extend library... See PR #11 

In my case I'm using it as a syntax for a self-closing block. It's in a library similar to `dominator`, so if you're familiar with that one and its signals-based API, you might imagine something like this following:
```
<
  {self.todos
    .signal_vec()
    .len()
    .map(|len| if len > 0 { Some(...) } else { None }}
/>
```

This would be an alternative to JSX's `{self.todos.length > 0 && ...}` syntax

Use case taken from another app, where you can drop into and out of an editing mode by double-clicking:
```
<{if self.in_edit_mode {
    "textarea"
  } else {
    "p"
  }}
  class={self.generate_class()}
>{self.text}
...```

**Comment by stoically** [comment]
Implemented in #11, thanks @gbj!

Published in v0.8.0-beta.2
